genesyscloud/util_retries.go

```diff
@@ -39,7 +40,8 @@ func withRetriesForReadCustomTimeout(ctx context.Context, timeout time.Duration,
 		errStringLower := strings.ToLower(fmt.Sprintf("%v", err))
 		if strings.Contains(errStringLower, "timeout while waiting for state to become") ||
 			strings.Contains(errStringLower, "context deadline exceeded") {
-			ctx, _ := context.WithTimeout(context.Background(), timeout)
+			ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
We should never ignore the cancel object and should explicitly do a defer cancel() on the returned function.  This was caught by go vet.